### PR TITLE
new package yt-dlp

### PR DESCRIPTION
**Pull request by saltedcoffii**
## Description
yt-dlp is a [youtube-dl](https://github.com/chromebrew/chromebrew/blob/master/packages/youtube_dl.rb) fork with additional patches and features. For example, yt-dlp supports downloading videos from TikTok, where youtube-dl errors.
- Updated multiple `py3_` packages herein to use `pip` during install.

## Additional information
I also took the liberty to update its dependencies to the latest version.

Works properly (tested in containers):
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/saltedcoffii/chromebrew.git CREW_TESTING_BRANCH=yt_dlp_2022.10.14 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/py3_brotlicffi.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class Py3_brotlicffi < Package
+  description 'Python bindings to the Brotli compression library'
+  homepage 'https://github.com/python-hyper/brotlicffi/'
+  @_ver = '1.0.9.2'
+  version @_ver
+  license 'MIT'
+  compatibility 'all'
+  source_url 'https://github.com/python-hyper/brotlicffi.git'
+  git_hashtag "v#{@_ver}"
+
```
Comment (uberhacker) [inline packages/py3_brotlicffi.rb:12]: Do we need binaries or does this build fairly quickly?

File: packages/py3_mutagen.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class Py3_mutagen < Package
+  description 'Python module for handling audio metadata'
+  homepage 'https://mutagen.readthedocs.io/'
+  @_ver = '1.46.0'
+  version @_ver
+  license 'GPL-2'
+  compatibility 'all'
+  source_url 'https://github.com/quodlibet/mutagen.git'
+  git_hashtag "release-#{@_ver}"
+
```
Comment (uberhacker) [inline packages/py3_mutagen.rb:12]: Needs binaries?

File: packages/yt_dlp.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  def self.build
+    system 'make'
+    system 'make', 'completions'
+  end
+
+  def self.install
+    system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    # Makefile is screwy
+    FileUtils.mv "#{CREW_DEST_PREFIX}/man" "#{CREW_DEST_MAN_PREFIX}"
```
Comment (uberhacker) [inline packages/yt_dlp.rb:29]: You'll want to do this:
```
FileUtils.mv "#{CREW_DEST_PREFIX}/man/man1" "#{CREW_DEST_MAN_PREFIX}"
```
The `man` subdirectory is already included in `CREW_DEST_MAN_PREFIX`.

**Review by uberhacker — CHANGES_REQUESTED** [review]
I forgot to mention this is your earlier new package commit but we should also update `tools/packages.yaml` with all the new packages we're adding.  What I've been doing with the `activity` line in `tools/packages.yaml` is:
```
none => no new releases in over a year
low => 1 or more releases in the last year but none in the last 3 months
medium => 1 or more releases in the last 3 months
high => 1 or more releases in each of the last 3 months
```
Also, the `url` is where you can download new releases.  Hopefully, that makes sense.

**Comment by saltedcoffii** [comment]
@uberhacker what is the purpose of packages.yaml?

**Comment by uberhacker** [comment]
> @uberhacker what is the purpose of packages.yaml?

The original intention was for update notifications using `urlwatch`.  A python script checks the url periodically for changes and reports new updates.  Ultimately, I'd like to get this process automated in Slack.

File: packages/yt_dlp.rb [hunk 4]
```diff
@@ -26,8 +26,8 @@ def self.build
   def self.install
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
     # Makefile is screwy
-    Dir.glob("#{CREW_DEST_PREFIX}/man/man*").each manfolder do
-      FileUtils.mv manfolder CREW_DEST_MAN_PREFIX
+    Dir.glob("#{CREW_DEST_PREFIX}/man/man*").each i do
+      FileUtils.mv i CREW_DEST_MAN_PREFIX
     end
```
Comment (uberhacker) [inline packages/yt_dlp.rb:31]: Easier?
```
FileUtils.mv Dir["#{CREW_DEST_PREFIX}/man/*"], CREW_DEST_MAN_PREFIX
```

File: packages/yt_dlp.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  def self.build
+    system 'make'
+    system 'make', 'completions'
+  end
+
+  def self.install
+    system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    # Makefile is screwy
+    Dir.glob("#{CREW_DEST_PREFIX}/man/man*").each manfolder do
```
Comment (supechicken) [inline packages/yt_dlp.rb:29]: This is not a valid ruby code...
```ruby
Dir.glob("#{CREW_DEST_PREFIX}/man/man*") do |manfolder|
```

Comment (saltedcoffii) [reply packages/yt_dlp.rb:29]: okay thank you

File: packages/py3_brotlicffi.rb (hunk 1 above)
Comment (saltedcoffii) [reply packages/py3_brotlicffi.rb:12]: binaries, it invokes a c compiler

File: packages/py3_mutagen.rb (hunk 2 above)
Comment (saltedcoffii) [reply packages/py3_mutagen.rb:12]: invokes a c compiler

File: packages/yt_dlp.rb (hunk 3 above)
Comment (saltedcoffii) [reply packages/yt_dlp.rb]: done.

**Comment by satmandu** [comment]
We also need packages.yaml updated for the new packages...

File: packages/py3_brotlicffi.rb (hunk 1 above)
Comment (satmandu) [reply packages/py3_brotlicffi.rb:12]: installs quickly.

File: packages/py3_mutagen.rb (hunk 2 above)
Comment (satmandu) [reply packages/py3_mutagen.rb:12]: installs quickly

File: packages/py3_brotlicffi.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    system "python -s -m pip install #{name.gsub('py3_', '')}", exception: false
+    @pip_files = `python -s -m pip show -f #{name.gsub('py3_', '')}`.chomp
+    @pip_files_base = @pip_files[/(?<=Location: ).*/, 0].concat('/')
+    @pip_files_lines = @pip_files[/(?<=Files:\n)[\W|\w]*/, 0].split
+    @pip_files_lines.each do |pip_file|
+      @pip_path = File.expand_path("#{@pip_files_base}#{pip_file}")
+      @destpath = "#{CREW_DEST_DIR.chomp('/')}#{@pip_path}"
+      FileUtils.install @pip_path, @destpath
+    end
+  end
```
Comment (uberhacker) [inline packages/py3_brotlicffi.rb:29]: Missing `def self.remove`?

File: packages/py3_certifi.rb [hunk 7]
```diff
@@ -3,33 +3,26 @@
...
+    system "python -s -m pip install #{name.gsub('py3_', '')}", exception: false
+    @pip_files = `python -s -m pip show -f #{name.gsub('py3_', '')}`.chomp
+    @pip_files_base = @pip_files[/(?<=Location: ).*/, 0].concat('/')
+    @pip_files_lines = @pip_files[/(?<=Files:\n)[\W|\w]*/, 0].split
+    @pip_files_lines.each do |pip_file|
+      @pip_path = File.expand_path("#{@pip_files_base}#{pip_file}")
+      @destpath = "#{CREW_DEST_DIR.chomp('/')}#{@pip_path}"
+      FileUtils.install @pip_path, @destpath
+    end
   end
```
Comment (uberhacker) [inline packages/py3_certifi.rb:27]: Missing `def self.remove`?

File: packages/py3_mutagen.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    system "python -s -m pip install #{name.gsub('py3_', '')}", exception: false
+    @pip_files = `python -s -m pip show -f #{name.gsub('py3_', '')}`.chomp
+    @pip_files_base = @pip_files[/(?<=Location: ).*/, 0].concat('/')
+    @pip_files_lines = @pip_files[/(?<=Files:\n)[\W|\w]*/, 0].split
+    @pip_files_lines.each do |pip_file|
+      @pip_path = File.expand_path("#{@pip_files_base}#{pip_file}")
+      @destpath = "#{CREW_DEST_DIR.chomp('/')}#{@pip_path}"
+      FileUtils.install @pip_path, @destpath
+    end
+  end
```
Comment (uberhacker) [inline packages/py3_mutagen.rb:27]: Missing `def self.remove`?

File: packages/py3_pycryptodomex.rb [hunk 9]
```diff
@@ -3,34 +3,26 @@
...
+    system "python -s -m pip install #{name.gsub('py3_', '')}", exception: false
+    @pip_files = `python -s -m pip show -f #{name.gsub('py3_', '')}`.chomp
+    @pip_files_base = @pip_files[/(?<=Location: ).*/, 0].concat('/')
+    @pip_files_lines = @pip_files[/(?<=Files:\n)[\W|\w]*/, 0].split
+    @pip_files_lines.each do |pip_file|
+      @pip_path = File.expand_path("#{@pip_files_base}#{pip_file}")
+      @destpath = "#{CREW_DEST_DIR.chomp('/')}#{@pip_path}"
+      FileUtils.install @pip_path, @destpath
+    end
   end
```
Comment (uberhacker) [inline packages/py3_pycryptodomex.rb:27]: Missing `def self.remove`?

File: packages/py3_websockets.rb [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    system "python -s -m pip install #{name.gsub('py3_', '')}", exception: false
+    @pip_files = `python -s -m pip show -f #{name.gsub('py3_', '')}`.chomp
+    @pip_files_base = @pip_files[/(?<=Location: ).*/, 0].concat('/')
+    @pip_files_lines = @pip_files[/(?<=Files:\n)[\W|\w]*/, 0].split
+    @pip_files_lines.each do |pip_file|
+      @pip_path = File.expand_path("#{@pip_files_base}#{pip_file}")
+      @destpath = "#{CREW_DEST_DIR.chomp('/')}#{@pip_path}"
+      FileUtils.install @pip_path, @destpath
+    end
+  end
```
Comment (uberhacker) [inline packages/py3_websockets.rb:27]: Missing `def self.remove`?

File: packages/py3_websockets.rb [hunk 11]
```diff
@@ -0,0 +1,28 @@
+require 'package'
+
+class Py3_websockets < Package
+  description ' Library for building WebSocket servers and clients in Python'
+  homepage 'https://websockets.readthedocs.io/'
+  @_ver = '11.0.3'
+  version "#{@_ver}-py3.11"
+  license 'BSD-3'
+  compatibility 'all'
+  source_url 'https://github.com/aaugustin/websockets.git'
```
Comment (uberhacker) [inline packages/py3_websockets.rb:10]: Does `source_url 'SKIP'` work?

File: packages/py3_pycryptodomex.rb [hunk 12]
```diff
@@ -3,34 +3,26 @@
 class Py3_pycryptodomex < Package
   description 'Pycryptodomex is a cryptographic library for Python.'
   homepage 'https://www.pycryptodome.org/'
-  @_ver = '3.11.0'
+  @_ver = '3.18.0'
   version "#{@_ver}-py3.11"
   license 'BSD and public-domain'
   compatibility 'all'
   source_url 'https://github.com/Legrandin/pycryptodome.git'
   git_hashtag "v#{@_ver}x"
```
Comment (uberhacker) [inline packages/py3_pycryptodomex.rb:11]: Does `source_url 'SKIP'` work?

File: packages/py3_mutagen.rb [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+class Py3_mutagen < Package
+  description 'Python module for handling audio metadata'
+  homepage 'https://mutagen.readthedocs.io/'
+  @_ver = '1.46.0'
+  version "#{@_ver}-py3.11"
+  license 'GPL-2'
+  compatibility 'all'
+  source_url 'https://github.com/quodlibet/mutagen.git'
+  git_hashtag "release-#{@_ver}"
```
Comment (uberhacker) [inline packages/py3_mutagen.rb:11]: Does `source_url 'SKIP'` work?

File: packages/py3_certifi.rb [hunk 14]
```diff
@@ -3,33 +3,26 @@
 class Py3_certifi < Package
   description 'Certifi provides Mozilla\'s CA Bundle.'
   homepage 'https://certifi.io/' # 403 Forbidden on 2021.05.08, still 403 Forbidden on 2021.10.13
-  @_ver = '2021.10.08'
+  @_ver = '2023.05.07'
   version "#{@_ver}-py3.11"
   license 'MPL-2.0'
   compatibility 'all'
   source_url 'https://github.com/certifi/python-certifi.git'
   git_hashtag @_ver
```
Comment (uberhacker) [inline packages/py3_certifi.rb:11]: Does `source_url 'SKIP'` work?

File: packages/py3_brotlicffi.rb [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+class Py3_brotlicffi < Package
+  description 'Python bindings to the Brotli compression library'
+  homepage 'https://github.com/python-hyper/brotlicffi/'
+  @_ver = '1.0.9.2'
+  version "#{@_ver}-py3.11"
+  license 'MIT'
+  compatibility 'all'
+  source_url 'https://github.com/python-hyper/brotlicffi.git'
+  git_hashtag "v#{@_ver}"
```
Comment (uberhacker) [inline packages/py3_brotlicffi.rb:11]: Does `source_url 'SKIP'` work?

File: packages/py3_websockets.rb (hunk 11 above)
Comment (satmandu) [reply packages/py3_websockets.rb]: Done

**Review by uberhacker — APPROVED** [review]

**Review by uberhacker — CHANGES_REQUESTED** [review]
Missing manifest filelists?

**Comment by satmandu** [comment]
> Missing manifest filelists?

manifest filelists have been added along with binaries...

**Review by uberhacker — APPROVED** [review]

**Pull request merged**